### PR TITLE
Add Server#getRoleColor(User) and User#getRoleColor(Server)

### PR DESCRIPTION
**Pull request by BoneyIsSpooky**
From todo_general

### CODE REVIEW

File: javacord-api/src/main/java/org/javacord/api/entity/server/Server.java [hunk 1]
```diff
@@ -520,6 +522,19 @@ default Role getEveryoneRole() {
...
+     *
+     * @param user The user.
+     * @return The color.
+     */
+    default Optional<Color> getRoleColorOf(User user) {
+        return user.getRoles(this).stream()
+                   .filter(role -> role.getColor().isPresent())
+                   .max(Comparator.comparingInt(Role::getPosition))
+                   .flatMap(Role::getColor);
+    }
```
Comment (Vampire) [inline javacord-api/src/main/java/org/javacord/api/entity/server/Server.java]: If you get a `List` you already have a defined order, in this case by position, so sorting is unnecessary.
This should be sufficient I think:
```java
        return user.getRoles(this).stream()
                   .filter(role -> role.getColor().isPresent())
                   .findFirst();
```

Comment (Vampire) [reply javacord-api/src/main/java/org/javacord/api/entity/server/Server.java]: It is also 4 lines, but it is faster, as it does not have to sort and thus does not have to go through all roles unnecessarily.

Comment (Vampire) [reply javacord-api/src/main/java/org/javacord/api/entity/server/Server.java]: Ah, forget what I said, the roles are sorted from smallest to biggest, so you would have to reverse the list first and for that you would have to create a new list first.

**Comment by Vampire** [comment]
Iirc we usually also then add `User#getRoleColorOf(Server server)` that basically just calls the other for convenience.

File: javacord-api/src/main/java/org/javacord/api/entity/user/User.java [hunk 2]
```diff
@@ -244,6 +245,17 @@ default String getDiscriminatedName() {
...
     }
 
+    /**
+     * Gets the displayed color of the user based on his roles in a server.
+     *
+     * @param server The server.
+     * @return The color.
+     * @see Server#getRoleColorOf(User)
+     */
+    default Optional<Color> getRoleColor(Server server) {
```
Comment (Vampire) [inline javacord-api/src/main/java/org/javacord/api/entity/user/User.java]: Besides that there is a PR from me that removes all "Of"s, is there a reason you name the method inconsistently?

Comment (BoneyIsSpooky) [reply javacord-api/src/main/java/org/javacord/api/entity/user/User.java]: The first method was named as it was due to it being listed that way in the todo. I can change one of them, if that is needed.

File: javacord-api/src/main/java/org/javacord/api/entity/server/Server.java [hunk 3]
```diff
@@ -520,6 +522,19 @@ default Role getEveryoneRole() {
                         .collect(Collectors.toList()));
     }
 
+    /**
+     * Gets the displayed color of a user based on his roles in the server.
```
Comment (Vampire) [inline javacord-api/src/main/java/org/javacord/api/entity/server/Server.java]: to stay consistent with other docs I'd probably say `on the server`

File: javacord-api/src/main/java/org/javacord/api/entity/user/User.java [hunk 4]
```diff
@@ -244,6 +245,17 @@ default String getDiscriminatedName() {
         return server.getRolesOf(this);
     }
 
+    /**
+     * Gets the displayed color of the user based on his roles in a server.
```
Comment (Vampire) [inline javacord-api/src/main/java/org/javacord/api/entity/user/User.java]: to stay consistent with other docs I'd probably say `on the given server`

Comment (BoneyIsSpooky) [reply javacord-api/src/main/java/org/javacord/api/entity/user/User.java]: `in the given server` is most consistent with the methods around it.

Comment (Vampire) [reply javacord-api/src/main/java/org/javacord/api/entity/user/User.java]: Hm, right, we need to improve consistency in the docs :-D

**Pull request merged**